src/NServiceBus.Hosting.Windows/GenericHost.cs

```diff
@@ -39,6 +39,7 @@ public GenericHost(IConfigureThisEndpoint specifier, string[] args, List<Type> d
             }
 
             profileManager = new ProfileManager(assembliesToScan, args, defaultProfiles);
+            startableAndStoppableBag = new StartableAndStoppableBag(assembliesToScan);
```

Comment:
If we move the registration to the feature (see below) you should be able to just iterate over the "scanned types" https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/EndpointConfiguration.cs#L207 ?